Replace fmt.Errorf with errors.New when there are no values to format…

natemurthy opened the pull request:
The `fmt.Errorf` function is just a convenience wrapper around `errors.New`. If you just want to create an error string with no interpolation why not just use the latter?

mikedanese committed:
godeps: bump gopkg.in/square/go-jose.v2

pickup https://github.com/square/go-jose/pull/179

mikedanese committed:
jwt: support opaque signer and push errors to token generator creation

Commit:
test: allow etcd to run on other ports.

running `make test-integration` with the `ETCD_PORT` option set should
work. Numerous integration test expect the correct URL to be populated
in the `KUBE_INTEGRATION_ETCD_URL` environment variable.

coffeepac committed:
make stackdriver logging tests not block e2e runs

brendandburns committed:
Make CredentialProvider config loading deterministic.

gambol99 committed:
Type Fix

- nothing much here just a typo fix found when copied and pasting into a kops PR https://github.com/kubernetes/kops/pull/5349

mtaufen committed:
Kubelet creates and manages node leases

This extends the Kubelet to create and periodically update leases in a
new kube-node-lease namespace. Based on [KEP-0009](https://github.com/kubernetes/community/blob/master/keps/sig-node/0009-node-heartbeat.md),
these leases can be used as a node health signal, and will allow us to
reduce the load caused by over-frequent node status reporting.

- add NodeLease feature gate
- add kube-node-lease system namespace for node leases
- add Kubelet option for lease duration
- add Kubelet-internal lease controller to create and update lease
- add e2e test for NodeLease feature
- modify node authorizer and node restriction admission controller
to allow Kubelets access to corresponding leases

Commit:
fix typos

transfering -> transferring

Commit:
dry-run: Allow dry-run flag to pass through if alpha enabled

etiennecoutaud committed:
typo s/thub/github/g

Commit:
clean unused variable

CallMeFoxie committed:
Change README links to https

dixudx committed:
add new GVK kubescheduler.config.k8s.io/v1alpha1.KubeSchedulerConfiguration

andyzhangx committed:
Enable dynamic azure disk volume limits

use API to get max disk num

use continue when got exception

add logging

add cache and unit test

fix comments

hanxiaoshuai committed:
refactor hard code in test/e2e/apimachinery/garbage_collector.go

dixudx committed:
golint failure

mortent committed:
Better error message when checking rollout status for StatefulSet with OnDelete strategy type

dixudx committed:
use external KubeSchedulerConfiguration

krzysztof-jastrzebski committed:
Reduce latency to node ready after CIDR is assigned.

jbartosik committed:
Improve HPA sample sanitization

After my previous changes HPA wasn't behaving correctly in the following
situation:

- Pods use a lot of CPU during initilization, become ready right after they initialize,
- Scale up triggers,
- When new pods become ready HPA counts their usage (even though it's not related to any work that needs doing),
- Another scale up, even though existing pods can handle work, no problem.

Commit:
dry-run: Create integration test

dixudx committed:
auto-generated

Commit:
Using a fixed set of locks, then we don't need to free unused locks
anymore.
See kubernetes/kubernetes/pull/66442 for discussions.

stewart-yu committed:
[kube-controller-manager]output flags in logical sections

stewart-yu committed:
[cloud-controller-manager]output flags in logical sections

shyamjvs committed:
Add timeout to MIG wait-until-stable operation

freehan committed:
Turn on PodReadinessGate by default

freehan committed:
add NodeConformance test for PodReadinessGate

Commit:
fix kubelet iptclient in ipv6 cluster

rosti committed:
kubeadm: use ClusterConfiguration in images.go

Replace the unnecessary use of InitConfiguration in images.go with
ClusterConfiguration. This changes the interfaces of the following functions:

- GetKubeControlPlaneImage
- GetEtcdImage
- GetAllImages

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

stewart-yu committed:
auto-generated file

stewart-yu committed:
skip print section flags if no flags

sttts committed:
delegated authz: add AlwaysAllowPaths mechanism to exclude e.g. /healthz

neolit123 committed:
add labels to kubectl OWNERS files

louislepper committed:
Make error messages more helpful in some e2e tests

Related to issues 34059 and 10322

jbartosik committed:
Make HPA more configurable

Duration of initialization taint on CPU and window of initial readiness
setting controlled by flags.

Adding API violation exceptions following example of e50340ee238

louislepper committed:
Output more context with the log messages, and general style tweaks

Commit:
remove rescheduler

krzyzacy committed:
handle recover from panic

Commit:
diff: Fix crash when remote object doesn't exist

Since we're saving nil in an interface rather than the implementation,
we can't compare to nil to check if the remote object exists or
not. Change the struct to save in the implementation.

shyamjvs committed:
Parallelize kubemark cluster creation

tallclair committed:
WIP: RC pods

tallclair committed:
Generated code

Commit:
Replace fmt.Errorf with errors.New when there are no values to format string

Commit:
Replace fmt.Errorf with errors.New when there are no values to format string

FengyunPan2 committed:
Make OpenStack cloud provider report a node hostname address

Related to: #67714
Cloud-provider-reported addresses are authoritative in 1.12, this
preserves default behavior that used the internal dns name as
the "Hostname" address on Node status.

neolit123 commented:
the `fmt.Errorf` -> `errors.New` change is not bad it has performance implications.
but the PR touches other areas especially in kubeadm, please rebase your commits against the latest master.

neolit123 commented:
/release-note-none

tizhou86 committed:
Add unit test cases for scheduler/util.

tizhou86 committed:
Fix golint error under scheduler/algorithm/priorities.

tizhou86 committed:
Fix golint error under scheduler/factory.

houjun41544 committed:
Complement unit test case TestNodesWherePreemptionMightHelp for scheduler/core

nikhita committed:
apiextensions: remove area/custom-resources label from OWNERS

Sometimes changes in apiextensions-apiserver are not really related
to custom resources but just general api-machinery.

Adding the label to such PRs makes triaging using the label hard.

shyamjvs committed:
Re-calibrate load test throughputs

mborsz committed:
WaitForAllNodesSchedulable should check taints as well

shyamjvs committed:
Remove second round of scaling in load test

natemurthy commented:
@neolit123 Rebased. Any suggestions as to what I should put for release note?

Btw, this is approximately how many other `fmt.Errorf` expressions I could replace with `errors.New`:
```
nmurthy:kubernetes% rg "return fmt.Errorf\(\"(.*?)\"\)" | wc -l
    1692
```

Commit:
Merge pull request #67748 from FengyunPan2/report-hostname

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Make OpenStack cloud provider report a node hostname address

Related to: #67714
Cloud-provider-reported addresses are authoritative in 1.12, this
preserves default behavior that used the internal dns name as
the "Hostname" address on Node status.

**Release note**:

```release-note
The OpenStack cloud provider now reports a `Hostname` address type for nodes
```

Commit:
Merge pull request #67769 from shyamjvs/remove-second-scaling-round-in-load-test

Automatic merge from submit-queue (batch tested with PRs 67748, 67769). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Remove second round of scaling in load test

As it seems redundant to do this twice, not really adding much value. This should cut down ~1.5h in our 5k-node load tests, and helps a bit with https://github.com/kubernetes/test-infra/issues/8348.

/cc @wojtek-t 

```release-note
NONE
```

neolit123 commented:
> @neolit123 Rebased. 

thanks but it seems that it still touches other code?

> Any suggestions as to what I should put for release note?

given this isn't user facing there is no need for a release note.

> nmurthy:kubernetes% rg "return fmt.Errorf\(\"(.*?)\"\)" | wc -l
    1692

these are bug numbers, but the problem is that if you submit it all in as a single PR it might take _a while_ before it's merged, unless a top level approver comes by.

Commit:
Merge pull request #67031 from krzysztof-jastrzebski/node_startup

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Reduce latency to node ready after CIDR is assigned.

This adds code to execute an immediate runtime and node status update when the Kubelet sees that it has a CIDR, which significantly decreases the latency to node ready.

```release-note
Speed up kubelet start time by executing an immediate runtime and node status update when the Kubelet sees that it has a CIDR.
```

Commit:
Merge pull request #67406 from freehan/pod-ready

Automatic merge from submit-queue (batch tested with PRs 67031, 67406). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Pod Ready++ cont

```release-note
Turn on PodReadinessGate by default
```

ref: https://github.com/kubernetes/community/blob/master/keps/sig-network/0007-pod-ready%2B%2B.md

Commit:
Merge pull request #67687 from Lion-Wei/remote-reschrduler

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

remove rescheduler since scheduling DS pods by default scheduler is moving to beta

**What this PR does / why we need it**:

remove rescheduler since scheduling DS pods by default scheduler is moving to beta

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #64725

**Special notes for your reviewer**:

**Release note**:
```release-note
Remove rescheduler since scheduling DS pods by default scheduler is moving to beta.
```

Commit:
Merge pull request #67548 from neolit123/owners-kubectl

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Add labels to kubectl OWNERS files

**What this PR does / why we need it**:

This change makes it possible to automatically add the two labels: `area/kubectl` and `sig/cli` to PRs that touch the paths in question.

this already exists for kubeadm:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/OWNERS#L17-L19

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
refs https://github.com/kubernetes/community/issues/1808

**Special notes for your reviewer**:
none

**Release note**:

```release-note
NONE
```
/area kubectl
@kubernetes/sig-cli-pr-reviews 
/cc @cblecker @tpepper

tallclair committed:
Address @thockin feedback

tallclair committed:
regenerate files

Commit:
Merge pull request #65247 from gambol99/typo_fixup

Automatic merge from submit-queue (batch tested with PRs 65247, 63633, 67425). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Typo Fix

- nothing much here just a typo fix found when copying and pasting into a kops [#PR5349](https://github.com/kubernetes/kops/pull/5349)

Commit:
Merge pull request #63633 from misterikkit/hacktest

Automatic merge from submit-queue (batch tested with PRs 65247, 63633, 67425). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

test: allow etcd to run on other ports.

running `make test-integration` with the `ETCD_PORT` option set should
work. Numerous integration test expect the correct URL to be populated
in the `KUBE_INTEGRATION_ETCD_URL` environment variable.



**What this PR does / why we need it**:
This makes the ETCD_PORT option work correctly.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
I expect nobody is using the ETCD_PORT option, but I noticed that it is broken.

**Release note**:

```release-note
NONE
```
/sig testing

Commit:
Merge pull request #67425 from Lion-Wei/kubelet-ipv6

Automatic merge from submit-queue (batch tested with PRs 65247, 63633, 67425). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

fix kubelet iptclient in ipv6 cluster

**What this PR does / why we need it**:
Kubelet uses "iptables" instead of "ip6tables" in an ipv6-only cluster. This causes failed traffic for type: LoadBalancer services (and probably a lot of other problems).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #67398 

**Special notes for your reviewer**:


**Release note**:
```release-note
NONE
```

Commit:
Merge pull request #67707 from apelisse/fix-diff-404

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

diff: Fix crash when remote object doesn't exist

Since we're saving nil in an interface rather than the implementation,
we can't compare to nil to check if the remote object exists or
not. Change the struct to save in the implementation.



**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #62420 from mikedanese/jose-bump

Automatic merge from submit-queue (batch tested with PRs 67707, 62420). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

svcacct: support jose.OpaqueSigner and push errors to token generator creation

```release-note
NONE
```

Commit:
Merge pull request #67737 from tallclair/runtimeclass-podapi

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Runtimeclass podapi

Adds (but doesn't implement) the RuntimeClass Pod API proposed in https://github.com/kubernetes/community/blob/master/keps/sig-node/0014-runtime-class.md

Based on the result (lazy consensus) of the alpha fields discussion on sig-architecture: https://groups.google.com/forum/#!topic/kubernetes-sig-architecture/y9FulL9Uq6A

For https://github.com/kubernetes/features/issues/585

**Special notes for your reviewer**:

**Release note**:
```release-note
RuntimeClass is a new API resource for defining different classes of runtimes that may be used to run containers in the cluster. Pods can select a RunitmeClass to use via the RuntimeClassName field. This feature is in alpha, and the RuntimeClass feature gate must be enabled in order to use it.
```

/sig node
/kind api-change
/priority important-soon
/milestone v1.12

Commit:
Merge pull request #66983 from mortent/BetterRolloutStatusMsgForStatefulSet

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Better error message when checking rollout status for StatefulSet wit…

…h OnDelete strategy type



**What this PR does / why we need it**: The error message when checking the rollout status for a StatefulSet with the OnDelete strategy type can be confusing (ref #64500). It gives the impression that something has gone wrong when the issue is simply that there is no rollout status. The error message is updated to use similar language as for DaemonSet in the same situation.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:

```release-note
Improved error message when checking the rollout status of StatefulSet with OnDelete strategy type
```

k82cn commented:
/ok-to-test

neolit123 commented on cmd/kubeadm/app/cmd/upgrade/node.go:
```diff
@@ -108,59 +95,10 @@ func NewCmdUpgradeNodeConfig() *cobra.Command {
-	cmd.Flags().StringVar(&flags.advertiseAddress, "apiserver-advertise-address", flags.advertiseAddress, "If the node is joining as a master, the IP address the API Server will advertise it's listening on.")
-	cmd.Flags().StringVar(&flags.nodeName, "node-name", flags.nodeName, "Specify the node name.")
-	return cmd
-}
```
[1]

neolit123 commented on cmd/kubeadm/app/cmd/upgrade/node.go:
```diff
@@ -219,43 +157,3 @@ func printFilesIfDryRunning(dryRun bool, kubeletDir string) error {
-
-	fmt.Println("[upgrade] The control plane instance for this node was successfully updated!")
-	return nil
-}
```
^ [1]

neolit123 commented on cmd/kubeadm/app/phases/upgrade/postupgrade.go:
```diff
@@ -108,7 +108,7 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 
 	// Rotate the kube-apiserver cert and key if needed
-	if err := BackupAPIServerCertIfNeeded(cfg, dryRun); err != nil {
+	if err := backupAPIServerCertIfNeeded(cfg, dryRun); err != nil {
```
[1]

neolit123 commented on cmd/kubeadm/app/phases/upgrade/postupgrade.go:
```diff
@@ -172,8 +172,7 @@ func upgradeToSelfHosting(client clientset.Interface, cfg *kubeadmapi.InitConfig
 
-// BackupAPIServerCertIfNeeded rotates the kube-apiserver certificate if older than 180 days
-func BackupAPIServerCertIfNeeded(cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
+func backupAPIServerCertIfNeeded(cfg *kubeadmapi.InitConfiguration, dryRun bool) error {
```
[1]

neolit123 commented on cmd/kubeadm/app/phases/upgrade/prepull.go:
```diff
@@ -159,11 +160,6 @@ func buildPrePullDaemonSet(component, image string) *apps.DaemonSet {
-				MatchLabels: map[string]string{
-					"k8s-app": addPrepullPrefix(component),
-				},
-			},
```
[1]

neolit123 commented on cmd/kubeadm/app/phases/upgrade/staticpods.go:
```diff
@@ -185,29 +185,28 @@ func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticP
+		}
+		if component == constants.KubeAPIServer {
+			if err := certsphase.KubeadmCertEtcdAPIClient.CreateFromCA(cfg, caCert, caKey); err != nil {
+				return fmt.Errorf("failed to upgrade the %s %s-client certificate and key: %v", constants.KubeAPIServer, constants.Etcd, err)
```
[1]

neolit123 commented on cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go:
```diff
@@ -31,10 +31,10 @@ import (
-	// or during upgrade nodes
-	NodesKubeadmConfigClusterRoleName = "kubeadm:nodes-kubeadm-config"
+	// phase for additional master nodes
+	BootstrapDiscoveryClusterRoleName = "kubeadm:bootstrap-discovery-kubeadm-config"
```
[1]

neolit123 commented on cmd/kubeadm/app/cmd/upgrade/node.go:
```diff
@@ -63,13 +58,6 @@ type nodeUpgradeFlags struct {
-	nodeName         string
-	dryRun           bool
-}
-
```
[1]

neolit123 reviewed (changes requested):
/hold
to clarify - please remove code changes unrelated to the PR (marked with [1]). thanks!

liggitt committed:
Add verify script for staging repo metadata files

Commit:
Merge pull request #67758 from tizhou86/newPR2

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Fix golint error under scheduler/algorithm/priorities.

**What this PR does / why we need it**:
Fix golint error under scheduler/algorithm/priorities.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #65571 from max8899/patch-1

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

fix typo

**What this PR does / why we need it**:

fix typo:
transfering -> transferring

**Release note**:
```release-note
NONE
```

Commit:
Merge pull request #65858 from etiennecoutaud/typo-changelog-1.11

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Typo in changelog 1.11

There is no small PR 😄 !

nikhita committed:
api: change owners from apimachinery to architecture

fabriziopandini committed:
kubeadm config move ControlPlaneEndpoint to ClusterConfiguration

fabriziopandini committed:
autogenerated

mborsz committed:
Revert "Parallelize RC deletion in density test"

Commit:
Merge pull request #67706 from shyamjvs/speed-up-kubemark-setup

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Parallelize kubemark cluster creation

To help with https://github.com/kubernetes/test-infra/issues/8348 (should hopefully cut down ~3mins - we're already down to 1h15m from earlier)

/cc @wojtek-t 

```release-note
NONE
```

Commit:
Merge pull request #66336 from CallMeFoxie/https-readme

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Change README links to https

**What this PR does / why we need it**:
Changes links referenced in README.md to https. No visual changes to people, just more safety.

```release-note
NONE
```

Commit:
Fix error link in comment

Pingan2017 committed:
del unused func DefaultEventFilterFunc

junxu committed:
Fix the "dirname: missing operand" error of "make test"

fabriziopandini committed:
kubeadm - deprecate feature-gates HighAvailability, SelfHosting, CertsInSecrets

Commit:
Merge pull request #67377 from shyamjvs/add-mig-startup-timeout

Automatic merge from submit-queue (batch tested with PRs 67377, 67801). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Add timeout to MIG wait-until-stable operation

Fixes https://github.com/kubernetes/kubernetes/issues/67374

/cc @wojtek-t @mborsz 
/sig cluster-lifecycle

```release-note
NONE
```

Commit:
Merge pull request #67801 from mborsz/revert-67712-parallelize-density-test-deletion

Automatic merge from submit-queue (batch tested with PRs 67377, 67801). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Revert "Parallelize RC deletion in density test"

Reverts kubernetes/kubernetes#67712

2k test has failed because of this: even if we trigger removal in parallel, controller manager is deleting replicacontrollers sequentially:
```
I0823 13:35:41.906] Aug 23 13:35:41.906: INFO: Terminating { ReplicationController} density43650-0-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 1m0.00118401s
I0823 13:36:42.021] Aug 23 13:36:42.020: INFO: Terminating { ReplicationController} density43650-14-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 2m0.00202238s
I0823 13:37:42.056] Aug 23 13:37:42.055: INFO: Terminating { ReplicationController} density43650-8-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 3m0.004624139s
I0823 13:38:42.092] Aug 23 13:38:42.092: INFO: Terminating { ReplicationController} density43650-5-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 4m0.000797258s
I0823 13:39:32.095] Aug 23 13:39:32.095: INFO: Terminating { ReplicationController} density43650-4-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 4m50.000689353s
I0823 13:40:32.206] Aug 23 13:40:32.206: INFO: Terminating { ReplicationController} density43650-11-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 5m50.000731007s
I0823 13:41:32.206] Aug 23 13:41:32.206: INFO: Terminating { ReplicationController} density43650-7-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 6m50.000648956s
I0823 13:42:32.194] Aug 23 13:42:32.194: INFO: Terminating { ReplicationController} density43650-12-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 7m50.001155426s
I0823 13:43:32.282] Aug 23 13:43:32.282: INFO: Terminating { ReplicationController} density43650-10-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 8m50.000739417s
I0823 13:44:16.434] W0823 13:44:16.433780    7517 reflector.go:270] k8s.io/kubernetes/test/utils/pod_store.go:56: watch of *v1.Pod ended with: too old resource version: 624915 (659380)
I0823 13:44:32.282] Aug 23 13:44:32.282: INFO: Terminating { ReplicationController} density43650-9-584f5bb0-a6d6-11e8-b145-0a580a3c6f08 pods took: 9m50.000486689s
I0823 13:44:42.283] Aug 23 13:44:42.282: INFO: Unexpected error occurred: error while waiting for pods to become inactive density43650-2-584f5bb0-a6d6-11e8-b145-0a580a3c6f08: timed out waiting for the condition
I0823 13:44:42.331] Aug 23 13:44:42.331: INFO: Unexpected error occurred: error while waiting for pods to become inactive density43650-1-584f5bb0-a6d6-11e8-b145-0a580a3c6f08: timed out waiting for the condition
I0823 13:44:42.333] Aug 23 13:44:42.332: INFO: Unexpected error occurred: error while waiting for pods to become inactive density43650-6-584f5bb0-a6d6-11e8-b145-0a580a3c6f08: timed out waiting for the condition
I0823 13:44:42.334] Aug 23 13:44:42.333: INFO: Unexpected error occurred: error while waiting for pods to become inactive density43650-13-584f5bb0-a6d6-11e8-b145-0a580a3c6f08: timed out waiting for the condition
I0823 13:44:42.437] Aug 23 13:44:42.436: INFO: Unexpected error occurred: error while waiting for pods to become inactive density43650-3-584f5bb0-a6d6-11e8-b145-0a580a3c6f08: timed out waiting for the condition
```

Commit:
Merge pull request #67762 from nikhita/remove-area-customresources-label

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

apiextensions: remove area/custom-resources label from OWNERS

Sometimes changes in apiextensions-apiserver are not really related to custom resources but just general api-machinery (or worse godep updates). Adding the label `area/custom-resources` to such PRs makes triaging using the label hard.

Adding the label manually is much cleaner.

See https://github.com/kubernetes/kubernetes/pull/67753 for an example.

/assign sttts 


**Release note**:

```release-note
NONE
```

humblec committed:
Remove provisioner config from log message.

Signed-off-by: hchiramm <hchiramm@redhat.com>

deads2k committed:
controller expectations for deletion can be met by 404

Commit:
Merge pull request #67543 from sttts/sttts-auth-skip-paths

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

delegated authz: add AlwaysAllowPaths to option struct (defaulting to /healthz)

Add `AlwaysAllowPaths` field to delegated authz. These http paths are excluded from the authz chain.

Prerequisite for https://github.com/kubernetes/kubernetes/pull/64149 and https://github.com/kubernetes/kubernetes/pull/67069.

```release-note
Added --authorization-always-allow-paths to components doing delegated authorization to exclude certain HTTP paths like /healthz from authorization.
```

smarterclayton committed:
Tests that use CheckTestingNSDeletedExcept must be [Serial]

Otherwise, another e2e test could be running at the same time and cause
a namespace to get created, which fails this test.

Commit:
Merge pull request #67800 from nikhita/api-owners-change-to-architecture

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

api: change owners from apimachinery to architecture

Ref: https://github.com/kubernetes/kubernetes/pull/67672#issuecomment-415566887
Change in the sigs files in k/community: https://github.com/kubernetes/community/pull/2584

/sig api-machinery
/sig architecture

/cc dims
/assign lavalamp 

**Release note**:

```release-note
NONE
```

ddebroy committed:
Add DynamicProvisioningScheduling support for GCE PD and RePD

Signed-off-by: Deep Debroy <ddebroy@docker.com>

Commit:
Merge pull request #66916 from dixudx/kubeadm_scheduler_api

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Moving KubeSchedulerConfiguration from ComponentConfig API types to staging repos

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#528

**Special notes for your reviewer**:
/cc luxas timothysc 
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews 
**Release note**:

```release-note
Moving KubeSchedulerConfiguration from ComponentConfig API types to staging repos
```

Commit:
Merge pull request #67252 from jbartosik/metric-sanitization

Automatic merge from submit-queue (batch tested with PRs 66916, 67252, 67794, 67619, 67328). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Fix HPA sample sanitization

**What this PR does / why we need it**: @mwielgus pointed out a case when HPA fails as a result of my changes to HPA algorithm:
- Have pods that use a lot of CPU during initilization, become ready right after they initialize,
- Trigger a scale up,
- When new pods become ready will will count their usage (even though it's not related to any work that needs doing),
- This triggers another scale up, even though existing pods can handle work, no problem.

The fix is:
- Use all samples for non-cpu metrics.
- Only use CPU samples if:
  - Pod is ready and was started more than 2 minutes ago, or
  - Pod is unready and last readiness change happened more than 10s after it was started.

Reasoning behind this in: https://docs.google.com/document/d/1UdtYedhmCxjaJIQi6hwJMY0eHQQKxlVD8lSHZC1BPOA/edit

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:
```release-note
Replace scale up forbidden window with disregarding CPU samples collected when pod was initializing.
```

Commit:
Merge pull request #67794 from liggitt/verify-staging-metadata

Automatic merge from submit-queue (batch tested with PRs 66916, 67252, 67794, 67619, 67328). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Add verify script for staging repo metadata files

helps ensure we don't miss required files as new staging repos are added

```release-note
NONE
```

Commit:
Merge pull request #67619 from louislepper/issue/34059/10322-annotate-errors

Automatic merge from submit-queue (batch tested with PRs 66916, 67252, 67794, 67619, 67328). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Make error messages more helpful in some e2e tests

Related to issues 34059 and 10322



**What this PR does / why we need it**:
This PR makes error messages more helpful in some e2e tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This is my first time writing Go/Contributing to Kubernetes - let me know if I've missed anything! :-)

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67328 from cofyc/fix65113-2

Automatic merge from submit-queue (batch tested with PRs 66916, 67252, 67794, 67619, 67328). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Using a fixed set of locks, then we don't need to free unused locks anymore.

**What this PR does / why we need it**:

Using a fixed set of locks, then we don't need to free unused locks anymore.
See kubernetes/kubernetes/pull/66442 for discussions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #65113

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/assign @msau42 
/assign @thockin

Commit:
Merge pull request #67776 from shyamjvs/fix-load-test-throughput

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Re-calibrate load test throughputs

/cc @wojtek-t 

```release-note
NONE
```

Commit:
Merge pull request #67503 from rosti/kubeadm_clusterconfig_images

Automatic merge from submit-queue (batch tested with PRs 67776, 67503, 67679, 67786, 67830). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

 kubeadm: use ClusterConfiguration in images.go

**What this PR does / why we need it**:

This PR is the first in a series, targeting the replacement of InitConfiguration with ClusterConfiguration, when the former is not needed. Please, review only the last commit.

Replace the unnecessary use of InitConfiguration in images.go with ClusterConfiguration. This changes the interfaces of the following functions:

- GetKubeControlPlaneImage
- GetEtcdImage
- GetAllImages

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
refs kubernetes/kubeadm#963

**Special notes for your reviewer**:

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/kind enhancement
/assign @luxas
/assign @timothysc
/assign @fabriziopandini

Depends on:
- [X] #67441

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67679 from krzyzacy/recover

Automatic merge from submit-queue (batch tested with PRs 67776, 67503, 67679, 67786, 67830). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

handle recover from panic

The upgrade jobs like https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-stable1-beta-upgrade-master/700 are panic on this test and didn't print any output

add a recover to the sub go-routine.

/sig-instrumentation
/cc jayunit100

Commit:
Merge pull request #67786 from fabriziopandini/kubeadm-deprecate-featureflags

Automatic merge from submit-queue (batch tested with PRs 67776, 67503, 67679, 67786, 67830). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

kubeadm - deprecate feature-gates HighAvailability, SelfHosting, CertsInSecrets

**What this PR does / why we need it**:
As for sig discussion (see meeting notes - August 22 - 2018) we are going to block usage of feature gates HighAvailability, SelfHosting, CertsInSecrets for new clusters and block updates to v1.12 of existing clusters using such features.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` 
Fixes # https://github.com/kubernetes/kubeadm/issues/1058

**Special notes for your reviewer**:
I'm going to open issue to track code cleanup in v1.13 

**Release note**:
```release-note
kubeadm - feature-gates HighAvailability, SelfHosting, CertsInSecrets are now deprecated and can't be used anymore for new clusters. Update of cluster using above feature-gates flag is not supported
```
/sig cluster-lifecycle
/kind feature
/kind cleanup
/assign @timothysc
/cc

Commit:
Merge pull request #67830 from fabriziopandini/kubeadm-config-ControlPlaneEndpoint

Automatic merge from submit-queue (batch tested with PRs 67776, 67503, 67679, 67786, 67830). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

kubeadm config move ControlPlaneEndpoint to ClusterConfiguration

**What this PR does / why we need it**:
This PR moves `ControlPlaneEndpoint` from the `API` config struct to `ClusterConfiguration`.

This change is required as initial step for enabling management of more than one control plane instances in kubeadm

**Which issue(s) this PR fixes** :
refs https://github.com/kubernetes/kubeadm/issues/911, refs https://github.com/kubernetes/kubeadm/issues/963

**Special notes for your reviewer**:
just an appetizer, the main dish will be the next PR...

**Release note**:
```release-note
kubeadm: ControlPlaneEndpoint was moved from the API config struct to ClusterConfiguration
```

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/sig cluster-lifecycle
/area kubeadm
/kind api-change
/kind enhancement
/assign @luxas
/assign @timothysc
/cc @chuckha @rosti @neolit123 @liztio

Commit:
Merge pull request #67530 from ddebroy/ddebroy-gcepd1

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Add DynamicProvisioningScheduling support for GCE PD and RePD

**What this PR does / why we need it**:
This PR adds support for the DynamicProvisioningScheduling feature for GCE PD and RePD. With this in place, if VolumeBindingMode: WaitForFirstConsumer is specified in a GCE storageclass and DynamicProvisioningScheduling is enabled, GCE PD provisioner will use the selected node's LabelZoneFailureDomain as (1) the zone to provision a GCE PD volume in (2) one of the zones to provision GCE RePD volume in.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
E2E tests for DynamicProvisioningScheduling scenarios for GCE PD to follow 

**Release note**:
```release-note
none
```
/sig storage
/assign @msau42

liggitt committed:
Set paging feature correctly for aggregator and crd storage

Commit:
Merge pull request #67861 from liggitt/crd-chunking

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Set paging feature correctly for aggregator and crd storage

The paging feature was not being set correctly in storage options for the aggregator or custom resource servers.

/assign @smarterclayton 

```release-note
API paging is now enabled for custom resource definitions, custom resources and APIService objects
```

Commit:
Merge pull request #67759 from tizhou86/newPR3

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Fix golint error under scheduler/factory.

**What this PR does / why we need it**:
Fix golint error under scheduler/factory.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67822 from humblec/remove-config

Automatic merge from submit-queue (batch tested with PRs 67822, 67835). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Remove provisioner config from log message.

Signed-off-by: hchiramm <hchiramm@redhat.com>
```
release-note-none

```

Commit:
Merge pull request #67835 from smarterclayton/serial

Automatic merge from submit-queue (batch tested with PRs 67822, 67835). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Tests that use CheckTestingNSDeletedExcept must be [Serial]

Commit:
Merge pull request #67826 from deads2k/controller-03-missingisgone

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

controller expectations for deletion can be met by 404

A controller asks pod control to delete a pod because it wants the pod to be gone.  It doesn't really care if the imperative delete action itself succeeds.  When the pod is already gone (404), then the desire of the controller is met.

Since the pods themselves are cache driven, you can hit this condition more than you may like. See https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L582 as an example.

@kubernetes/sig-apps-bugs 
/assign @janetkuo @tnozicka 


```release-note
latent controller caches no longer cause repeating deletion messages for deleted pods
```

Commit:
Merge pull request #64564 from coffeepac/stackdriver-tests-non-blocking

Automatic merge from submit-queue (batch tested with PRs 64564, 65043). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

make stackdriver logging tests not block e2e runs

**What this PR does / why we need it**:  prevents Stackdriver logging test failures from failing an e2e run

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #65043 from brendandburns/todo4

Automatic merge from submit-queue (batch tested with PRs 64564, 65043). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Make CredentialProvider config loading deterministic.

@mattmoor fixed an old TODO :)

Commit:
Merge pull request #67805 from linyouchong/pr-0824

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Fix error link in comment

**What this PR does / why we need it**:
Fix error link in comment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
NONE
**Release note**:
```release-note
NONE
```

/sig node

Commit:
Merge pull request #67816 from junxu/make_test_dirname_err

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Fix "hack/make-rules/test.sh"

**What this PR does / why we need it**:
For “make test”, it will output "dirname: missing operand".

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

Commit:
Merge pull request #67760 from houjun41544/20180823

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

 Complement unit test case TestNodesWherePreemptionMightHelp for scheduler/core

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

tizhou86 committed:
Add unit test cases for scheduler/algorithm/predicates.

Commit:
Merge pull request #66257 from mtaufen/node-lease

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Kubelet creates and manages node leases

This extends the Kubelet to create and periodically update leases in a
new kube-node-lease namespace. Based on [KEP-0009](https://github.com/kubernetes/community/blob/master/keps/sig-node/0009-node-heartbeat.md),
these leases can be used as a node health signal, and will allow us to
reduce the load caused by over-frequent node status reporting.

- add NodeLease feature gate
- add kube-node-lease system namespace for node leases
- add Kubelet option for lease duration
- add Kubelet-internal lease controller to create and update lease
- add e2e test for NodeLease feature

I would like to determine a standard policy for lease renewal frequency, based on the configured lease duration, so that we don't need to expose frequency as an additional knob. The renew interval is currently calculated as 1/3 of the lease duration.

```release-note
kubelet: Users can now enable the alpha NodeLease feature gate to have the Kubelet create and periodically renew a Lease in the kube-node-lease namespace. The lease duration defaults to 40s, and can be configured via the kubelet.config.k8s.io/v1beta1.KubeletConfiguration's NodeLeaseDurationSeconds field.
```

/cc @wojtek-t @liggitt

Commit:
Merge pull request #67750 from tizhou86/newUnitTest9

Automatic merge from submit-queue (batch tested with PRs 66257, 67750). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Add unit test cases for scheduler/util.

**What this PR does / why we need it**:
Add unit test cases for scheduler/util for more code coverage.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

fabriziopandini committed:
kubeadm config add support for more than one APIEndpoint

fabriziopandini committed:
autogenerated

Commit:
Merge pull request #66322 from tanshanshan/fixunused

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

ignore unused variable

**What this PR does / why we need it**:
ignore unused variable 

unused err
https://github.com/kubernetes/kubernetes/blob/162655f42f79e03a0041f7d2dfb1a818be244ef7/pkg/cloudprovider/providers/openstack/openstack_routes_test.go#L89

and
unused instanceId
https://github.com/kubernetes/kubernetes/blob/162655f42f79e03a0041f7d2dfb1a818be244ef7/pkg/cloudprovider/providers/photon/photon_test.go#L143

found in #66303 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:


**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #66660 from hanxiaoshuai/cleanup0726

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

refactor hard code in test/e2e/apimachinery/garbage_collector.go

**What this PR does / why we need it**:
refactor hard code in test/e2e/apimachinery/garbage_collector.go
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67766 from mborsz/taints

Automatic merge from submit-queue (batch tested with PRs 67766, 67642, 67772). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

WaitForAllNodesSchedulable should check taints as well

**What this PR does / why we need it**:
In https://github.com/kubernetes/kubernetes/issues/67597 we see a lot of cases when test starts before not-ready and network-unavailable taints are removed from the nodes while they already have correct conditions.
This change makes sure that we wait for both.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

/assign @wojtek-t

Commit:
Merge pull request #67642 from tizhou86/newUnitTest7

Automatic merge from submit-queue (batch tested with PRs 67766, 67642, 67772). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Add unit test cases for scheduler/algorithm/predicates.

**What this PR does / why we need it**:
Add unit test cases for scheduler/algorithm/predicates for more code coverage.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67772 from andyzhangx/azuredisk-volumelimits2

Automatic merge from submit-queue (batch tested with PRs 67766, 67642, 67772). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Enable dynamic azure disk volume limits

**What this PR does / why we need it**:
Enable dynamic azure disk volume limits,
This is an azure cloud provider implementation related to feature: [Dynamic Maximum volume count](https://github.com/kubernetes/features/issues/554)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66269

**Special notes for your reviewer**:
This PR use `az.VirtualMachineSizesClient.List` to list all vm sizes under region, match vm size with current node size, and then got `MaxDataDiskCount`, the `GetVolumeLimits` happens in kubelet and will return `attachable-volumes-azure-disk` in node status as following example:
```
agentpool-22082114-0
    ...
    allocatable:
      attachable-volumes-azure-disk: "8"
      cpu: "2"
      ephemeral-storage: "28043041951"
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 7034772Ki
      pods: "30"
```

**Release note**:

```
Enable dynamic azure disk volume limits
```

/sig azure
/kind feature

Commit:
Merge pull request #67362 from stewart-yu/stewart-controller-manager-codeclean#02

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

*-controller-manager: output flags in logical sections

![image](https://user-images.githubusercontent.com/30410021/44071436-8e619e72-9fbb-11e8-84c5-72c1ff9767b6.png)

```
Print *-controller-manager --help flag help in sections.
```

Commit:
Merge pull request #67256 from apelisse/dry-run-integration

Automatic merge from submit-queue (batch tested with PRs 67362, 67256, 67809). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

Dry run integration

Implement an integration test for dry-run. Also, this turns on the knob to allow dry-run requests, so let's be careful.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67809 from Pingan2017/dafaultfitler

Automatic merge from submit-queue (batch tested with PRs 67362, 67256, 67809). If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

del unused func DefaultEventFilterFunc

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

Commit:
Merge pull request #67832 from fabriziopandini/kubeadm-config-APIEndpoint

Automatic merge from submit-queue. If you want to cherry-pick this change to another branch, please follow the instructions <a href="https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md">here</a>.

kubeadm config add support for more than one APIEndpoint

**What this PR does / why we need it**:
This PR completes the changes in kubeadm for management of more than one control plane instances introducing the possibility to configure more than one APIEndpoints

**Which issue(s) this PR fixes** :
refs https://github.com/kubernetes/kubeadm/issues/911, refs https://github.com/kubernetes/kubeadm/issues/963

**Special notes for your reviewer**:
Depends on:
- [x] https://github.com/kubernetes/kubernetes/pull/67830

**Release note**:
```release-note
kubeadm: The kubeadm configuration now support definition of more than one control plane instances with their own APIEndpoint. The APIEndpoint for the "bootstrap" control plane instance should be defined using `InitConfiguration.APIEndpoint`, while the APIEndpoints for additional control plane instances should be added using `JoinConfiguration.APIEndpoint`.  
```

/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/sig cluster-lifecycle
/area kubeadm
/kind api-change
/kind enhancement
/assign @luxas
/assign @timothysc
/cc @chuckha @rosti @neolit123 @liztio

Commit:
Resolve merge conflicts

natemurthy commented:
I recently ran a
```
git fetch upstream
git rebase upstream/master
git push -f origin master
```
I don't understand where all these other code changes came from.

natemurthy commented:
This isn't worth the effort to me any more to constantly keep my fork up-to-date with master upstream.

neolit123 commented:
/hold
/uncc
